frontend/src/components/App/PluginSettings/pluginSettingsDetail.tsx

```diff
@@ -0,0 +1,180 @@
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Stack from '@mui/material/Stack';
+import withStyles from '@mui/styles/withStyles';
+import _ from 'lodash';
+import { isValidElement, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { deletePlugin } from '../../../lib/k8s/apiProxy';
+import { ConfigStore } from '../../../plugin/configStore';
+import { PluginInfo, reloadPage } from '../../../plugin/pluginsSlice';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+import NotFoundComponent from '../../404';
+import { SectionBox } from '../../common';
+import { ConfirmDialog } from '../../common/Dialog';
+import ErrorBoundary from '../../common/ErrorBoundary';
+
+export default function PluginSettingsDetail() {
+  const pluginSettings = useTypedSelector(state => state.plugins.pluginSettings);
+  const { name } = useParams<{ name: string }>();
+  const decodedName = decodeURIComponent(name);
+  const plugin = pluginSettings.find(plugin => plugin.name === decodedName);
+  if (!plugin) {
+    return <NotFoundComponent />;
+  }
+
+  function handleDeleteConfirm() {
+    if (!plugin) {
+      return;
+    }
+    const name = plugin?.name.split('/').splice(-1)[0];
+    deletePlugin(name)
+      .then(() => {
+        // update the plugin list
+        const dispatch = useDispatch();
+        dispatch(reloadPage());
+      })
+      .finally(() => {
+        // redirect /plugins page
+        // history.push isn't updating the plugins list so we're using window.location.pathname
+        window.location.pathname = '/settings/plugins';
```

Comment:
Please check if this works both in the app and the web modes. As the router is different.

Reply:
Tested this, works in app too.